Auto Capitlization and Punctuation

AgamCS opened the pull request:
Automatically capitalizes the first word of every sentence and adds punctuation if there isn't any. (For real this time)

Phibonacci commented:
I do appreciate the pull request but I am going to ask you to make some adjustment.

`ClientSend.lua` is not really meant to contain logic, although some of its functions do it that's only because there is no other dedicated file for that. But most importantly I feel like this feature should be optional, like for the verb prefix.

- Move the code in a function that will be call in:
  - `ISChat.onMessagePacket` (`media\lua\client\tics\client\overriden\ISChat.lua` line 946)
       By creating a new variable from the `message` variable, for instance:
       ```lua
       local cleanMessage = message
       if TicsServerSettings ~= nil and TicsServerSettings['options']['capitalize'] == true then
           cleanMessage = CapitalizeAndPonctuate(message)`
       end
       ```
       before the following line:
       `local formattedMessage, parsedMessage = BuildMessageFromPacket(type, message, name, color, nil, disableVerb)`
       and propagate that new `cleanMessage` in the rest of the function instead of `message`.

  - Same thing in `ISChat.onRadioPacket` line 1131
- Make it optional so people can use the sandbox options to enable it:
  - Add a `Capitalize` variable in `media\sandbox-options.txt`
  - And an English description for it in `media\lua\shared\Translate\EN\Sandbox_EN.txt`
  - Add the sandbox variable to the settings in `media\lua\server\tics\server\ChatMessage.lua` in function `SetMessageTypeSettings` line 103, it will be loaded automatically on the client side in the `TicsServerSettings`

AgamCS commented:
Hey! Sorry about that. I wasn't exactly sure how to set it up, but I've modified the files as you asked and added the sandbox option. Let me know if anything else needs changing!

Phibonacci commented:
Thank you for being so patient. I just merged your changes to my last feature.

I am currently dealing with personal issue so I cannot make any promise on the exact date I will deploy the changes but I hope to be able to do that next week.
